cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.test.cpp

```diff
@@ -0,0 +1,430 @@
+                   base_p_x * v + base_op;
+    uint512_t quotient_by_modulus = uint_previous_accumulator * uint_x + uint_z_2 * uint_v_quarted +
+                                    uint_z_1 * uint_v_cubed + uint_p_y * uint_v_squared + uint_p_x * uint_v + uint_op -
+                                    uint512_t(remainder);
+
+    uint512_t quotient = quotient_by_modulus / uint512_t(Fp::modulus);
+    constexpr uint512_t MAX_CONSTRAINED_SIZE = uint512_t(1) << 254;
+    constexpr uint512_t MAX_Z_SIZE = uint512_t(1) << (NUM_LIMB_BITS * 2);
+    numeric::uint1024_t max_quotient =
+        (uint1024_t(MAX_CONSTRAINED_SIZE) * MAX_CONSTRAINED_SIZE * 3 + MAX_Z_SIZE * MAX_CONSTRAINED_SIZE * 2 + 4) /
+        modulus_u512;
+    info("Max quotient: ", max_quotient);
+    info("Max quotient range constraint: ", max_quotient.get_msb() + 1);
+
+    auto [remainder_0, remainder_1, remainder_2, remainder_3, remainder_prime] = base_element_to_bigfield(remainder);
+    std::array<Fr, 5> remainder_witnesses = { remainder_0, remainder_1, remainder_2, remainder_3, remainder_prime };
+    auto [quotient_0, quotient_1, quotient_2, quotient_3, quotient_prime] = uint512_t_to_bigfield(quotient);
+    std::array<Fr, 5> quotient_witnesses = { quotient_0, quotient_1, quotient_2, quotient_3, quotient_prime };
+
+    // We will divide by shift_2 instantly in the relation itself, but f
+    Fr low_wide_relation_limb_part_1 =
+        previous_accumulator_witnesses[0] * x_witnesses[0] + op + v_witnesses[0] * p_x_witnesses[0] +
+        v_squared_witnesses[0] * p_y_witnesses[0] + v_cubed_witnesses[0] * z_1_lo + v_quarted_witnesses[0] * z_2_lo +
+        quotient_witnesses[0] * neg_modulus_limbs[0] - remainder_witnesses[0]; // This covers the lowest limb
+    info("LW1:", low_wide_relation_limb_part_1);
+    Fr low_wide_relation_limb =
+        low_wide_relation_limb_part_1 +
+        (previous_accumulator_witnesses[1] * x_witnesses[0] + previous_accumulator_witnesses[0] * x_witnesses[1] +
+         v_witnesses[1] * p_x_witnesses[0] + p_x_witnesses[1] * v_witnesses[0] +
+         v_squared_witnesses[1] * p_y_witnesses[0] + v_squared_witnesses[0] * p_y_witnesses[1] +
+         v_cubed_witnesses[1] * z_1_lo + z_1_hi * v_cubed_witnesses[0] + v_quarted_witnesses[1] * z_2_lo +
+         v_quarted_witnesses[0] * z_2_hi + quotient_witnesses[0] * neg_modulus_limbs[1] +
+         quotient_witnesses[1] * neg_modulus_limbs[0] - remainder_witnesses[1]) *
+            shift_1; // And this covers the limb shifted by 68
+    for (auto& limb : quotient_witnesses) {
+        info("Q: ", limb);
+    }
+    // Treating accumulator as 254-bit constrained value
+    constexpr auto max_limb_size = (uint512_t(1) << NUM_LIMB_BITS) - 1;
+    constexpr auto shift_1_u512 = uint512_t(shift_1);
+    constexpr auto op_max_size = uint512_t(4);
+    constexpr uint512_t lwl_maximum_value = op_max_size + (max_limb_size * max_limb_size) * ((shift_1_u512 * 12) + 6);
```

Comment:
Just flagging that its not immediately clear to me what `lwl` stands for / refers to. Low wide limb?

Reply:
Changed